api/app/src/main/kotlin/packit/controllers/RoleController.kt

```diff
@@ -53,4 +55,38 @@ class RoleController(private val roleService: RoleService)
 
         return ResponseEntity.ok(mapOf("message" to "Permissions removed"))
     }
+
+    @GetMapping
+    fun getRoleNames(): ResponseEntity<List<String>>
+    {
+        return ResponseEntity.ok(roleService.getRoleNames())
+    }
+
+    @GetMapping("/complete")
+    fun getRolesWithRelationships(): ResponseEntity<List<RoleDto>>
+    {
+        val roles = roleService.getRolesWithRelationships()
+        return ResponseEntity.ok(roles.map { it.toDto() })
+    }
+
+    @GetMapping("/complete/usernames")
+    fun getUsernameRoles(): ResponseEntity<List<RoleDto>>
+    {
+        val roles = roleService.getRolesWithRelationships(true)
+        return ResponseEntity.ok(roles.map { it.toDto() })
+    }
+
+    @GetMapping("/complete/non-usernames")
+    fun getNonUsernameRoles(): ResponseEntity<List<RoleDto>>
+    {
+        val roles = roleService.getRolesWithRelationships(false)
+        return ResponseEntity.ok(roles.map { it.toDto() })
+    }
+
+    @GetMapping("/{roleName}")
+    fun getRole(@PathVariable roleName: String): ResponseEntity<RoleDto>
+    {
+        val role = roleService.getRole(roleName)
```

Comment:
Is this a role "WithRelationships" too? It's returning the same Dto type so I assume so. Maybe drop the "withRelationships" from the plural method in the service for consistency.